### ISSUE TITLE
Add `loss_func` parameter to Trainer

### ISSUE THREAD

**Issue by gcroci2** [DeepRank/deeprank-core#153]
In `NeuralNet` class, if task is regression the loss is set as `MSELoss()`, otherwise as `CrossEntropyLoss()`.

- The user should be able to set whatever loss function. Thus, `loss_func` parameter should be added as input to `NeuralNet`. We could list the main losses for the two tasks and allow the user to either:
  - select one loss among them (considering also `task` parameter)
  - implement his/her own loss function.
- The defaults can be  `MSELoss()` for regression and `CrossEntropyLoss()` for classification.

**Comment by gcroci2** [comment]
Blocked by #172
